driver/mloKthValueHost.hpp

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+
+#include <../test/ford.hpp>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloKthvalueFwdRunHost(const miopenTensorDescriptor_t inputDesc,
+                              const miopenTensorDescriptor_t outputDesc,
+                              const miopenTensorDescriptor_t indicesDesc,
+                              const Tgpu* input,
+                              Tcheck* output,
+                              size_t* indices,
+                              const size_t k,
+                              const uint64_t dim)
+{
+    size_t input_size = miopen::deref(inputDesc).GetElementSize();
+    size_t dim_size   = miopen::deref(inputDesc).GetLengths()[dim];
+    size_t dim_stride = miopen::deref(inputDesc).GetStrides()[dim];
+
+    auto input_tv             = miopen::get_inner_expanded_tv<5>(miopen::deref(inputDesc));
+    auto input_tv_without_dim = miopen::get_tv_without_dim<5>(input_tv, dim);
+    auto output_tv            = miopen::get_inner_expanded_tv<5>(miopen::deref(outputDesc));
+    auto indices_tv           = miopen::get_inner_expanded_tv<5>(miopen::deref(indicesDesc));
+
+    size_t num_slice = input_size / dim_size;
+
+    std::vector<float> elements;
+    std::vector<size_t> ids(dim_size);
+    for(size_t i = 0; i < dim_size; ++i)
+    {
+        ids[i] = i;
+    }
+
+    for(size_t slide_id = 0; slide_id < num_slice; ++slide_id)
```

Comment:
check typo: it should be `slice` instead of `slide`
```suggestion
    for(size_t slice_id = 0; slice_id < num_slices; ++slice_id)
```